### PR TITLE
Controls cleanup

### PR DESCRIPTION
**Pull request by muthuArivoli**
Taken from #184 and rebased onto current repo structure.

### CODE REVIEW

File: onboard/catkin_ws/src/controls/scripts/drc_utils.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+from tf.transformations import euler_from_quaternion, quaternion_multiply, quaternion_conjugate
+
+
+def get_directions():
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/controls/scripts/drc_utils.py:4]: Maybe think of a better word for this than directions, directions implies only xyz. Maybe get_axes?

File: onboard/catkin_ws/src/controls/scripts/drc_utils.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    return ['x', 'y', 'z', 'roll', 'pitch', 'yaw']
+
+
+def get_pose_topic(direction):
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/controls/scripts/drc_utils.py:8]: And then also change the name fed in to these places

File: onboard/catkin_ws/src/controls/scripts/drc_utils.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+from tf.transformations import euler_from_quaternion, quaternion_multiply, quaternion_conjugate
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/controls/scripts/drc_utils.py:1]: Lets name this just utils, or controls_utils. drc_utils implies that it is utils used throughout the codebase.

File: onboard/catkin_ws/src/controls/scripts/state_republisher.py [hunk 4]
```diff
@@ -3,71 +3,37 @@
 import rospy
 from std_msgs.msg import Float64
 from nav_msgs.msg import Odometry
-from tf.transformations import euler_from_quaternion
+import drc_utils as utils
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/controls/scripts/state_republisher.py:6]: Remember to change drc_utils here too

File: onboard/catkin_ws/src/controls/scripts/thruster.py [hunk 5]
```diff
@@ -16,14 +15,13 @@ def __init__(self, pos, rpy, flipped):
...
 
-        q = quaternion_from_euler(to_rad(rpy[0]), to_rad(rpy[1]), to_rad(rpy[2]), 'sxyz') # create a quaternion from euler rpy
+        # create a quaternion from euler rpy
+        q = quaternion_from_euler(np.deg2rad(rpy[0]), np.deg2rad(rpy[1]), np.deg2rad(rpy[2]), 'sxyz')
         # sxyz denotes a static (extrinsic) frame of reference.
         # https://github.com/matthew-brett/transforms3d/blob/master/transforms3d/euler.py
         # https://en.wikipedia.org/wiki/Euler_angles
 
-        self.force_hat = quat_vec_mult(q, [1, 0, 0]) # vector representing force of thruster in x,y,z directions
+        self.force_hat = quat_vec_mult(q, [1, 0, 0])  # vector representing force of thruster in x,y,z directions
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/controls/scripts/thruster.py:25]: Why hat in the name here?

File: onboard/catkin_ws/src/controls/scripts/thruster_controls.py [hunk 6]
```diff
@@ -99,63 +73,21 @@ def transform_twist(self, base_frame, target_frame, twist):
...
         self.t_allocs = self.tm.calc_t_allocs(self.pid_outputs_local)
 
-
-    def _on_x(self, x):
-        self.pid_outputs[0] = x.data
-        self.update_thruster_allocs()
-
-    def _on_y(self, y):
-        self.pid_outputs[1] = y.data
+    def _on(self, val, direction):
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/controls/scripts/thruster_controls.py:85]: on what? Give this method a more descriptive name

File: onboard/catkin_ws/src/static_transforms/launch/static_transforms.launch [hunk 7]
```diff
@@ -2,5 +2,5 @@
 	<node pkg="tf2_ros" type="static_transform_publisher" name="corner_link_publisher" args="0.1080 0.1014 0.1575 0 0 0 base_link corner_link" />
 	<node pkg="tf2_ros" type="static_transform_publisher" name="dvl_link_publisher" args="-0.3118 -0.1014 -0.3894 0 0 0 corner_link dvl_link" />
 	<node pkg="tf2_ros" type="static_transform_publisher" name="cameras_link_publisher" args="0.6 0.2 0.3 0 0 0 corner_link cameras_link" />
-	<node pkg="tf2_ros" type="static_transform_publisher" name="imu_link_publisher" args="-0.0450 -0.1013 -0.1282 0 0 3.1416 corner_link imu_link" />
+	<node pkg="tf2_ros" type="static_transform_publisher" name="imu_link_publisher" args="-0.0450 -0.1013 -0.1282 0 0 0 corner_link imu_link" />
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/static_transforms/launch/static_transforms.launch:5]: Why this change?

File: onboard/catkin_ws/src/controls/launch/controls.launch [hunk 8]
```diff
@@ -7,7 +7,7 @@
         <arg name="sim" value="$(arg sim)" />
     </include>
 
-    <!--node pkg="tf2_ros" type="static_transform_publisher" name="base_link_publisher" args="0 0 0 0 0 -0.7070727 0.7071408 odom base_link" /-->
+    <!--node pkg="tf2_ros" type="static_transform_publisher" name="base_link_publisher" args="0 0 0 0 0 0 1 odom base_link" /-->
```
Comment (davidAmiron) [inline onboard/catkin_ws/src/controls/launch/controls.launch:10]: Why this change?

File: onboard/catkin_ws/src/controls/scripts/drc_utils.py (hunk 1 above)
Comment (rabinowitz) [reply onboard/catkin_ws/src/controls/scripts/drc_utils.py]: fixed in 189a96e

File: onboard/catkin_ws/src/controls/scripts/drc_utils.py (hunk 2 above)
Comment (rabinowitz) [reply onboard/catkin_ws/src/controls/scripts/drc_utils.py]: fixed in 189a96e

File: onboard/catkin_ws/src/controls/scripts/drc_utils.py (hunk 3 above)
Comment (rabinowitz) [reply onboard/catkin_ws/src/controls/scripts/drc_utils.py]: fixed in 189a96e

File: onboard/catkin_ws/src/controls/scripts/state_republisher.py (hunk 4 above)
Comment (rabinowitz) [reply onboard/catkin_ws/src/controls/scripts/state_republisher.py]: fixed in 189a96e

File: onboard/catkin_ws/src/controls/scripts/thruster.py (hunk 5 above)
Comment (rabinowitz) [reply onboard/catkin_ws/src/controls/scripts/thruster.py]: comment clarified in 189a96e

File: onboard/catkin_ws/src/controls/scripts/thruster_controls.py (hunk 6 above)
Comment (rabinowitz) [reply onboard/catkin_ws/src/controls/scripts/thruster_controls.py]: clarified in 189a96e

File: onboard/catkin_ws/src/static_transforms/launch/static_transforms.launch (hunk 7 above)
Comment (rabinowitz) [reply onboard/catkin_ws/src/static_transforms/launch/static_transforms.launch:5]: was done during pool testing on march 3

File: onboard/catkin_ws/src/controls/launch/controls.launch (hunk 8 above)
Comment (rabinowitz) [reply onboard/catkin_ws/src/controls/launch/controls.launch:10]: reverting a change we'd previously made. that line is commented out just for testing at the moment, and will eventually be parameterized to work with our GUI once we define a better way to test controls.

**Comment by rabinowitz** [comment]
@davidAmiron Shaan tested this to make sure we didn't break anything, so we should be good to merge when you're ready.
